generators/app/index.js

```diff
@@ -178,70 +234,113 @@ module.exports = class extends Generator {
           }
         });
         fs.writeFileSync(shaMarker, commitSHA);
+        this._hideBusy();
 
-        this.log("Installing the plugin...");
-        spawn.sync(this.shouldUseYarn() ? "yarn" : "npm", ["install"], {
-          stdio: "ignore",
-          cwd: generatorPath,
-        });
+        // run yarn/npm install
+        if (this.options.verbose) {
+          this.log("Installing the plugin dependencies...");
+        }
+        this._showBusy(`  - Preparing "${generator.name}"`);
+        await new Promise(function(resolve, reject) {
+          const process = spawn(this._shouldUseYarn() ? "yarn" : "npm", ["install"], {
+            stdio: "ignore",
+            cwd: generatorPath,
+          }).on('exit', function (code) {
+            resolve(code);
+          }).on('error', function (err) {
+            reject(err);
+          });
+        }.bind(this));
+        this._hideBusy();
+        this.log("  - Ready!");
       }
     }
 
-    const yeoman = require("yeoman-environment");
-
+    // filter the local options and the help command
     const opts = Object.keys(this._options).filter(
       (optionName) =>
         !(generatorOptions.hasOwnProperty(optionName) || optionName === "help")
     );
 
+    // create the env for the plugin generator
+    const yeoman = require("yeoman-environment");
     const env = yeoman.createEnv(this.args, opts);
 
+    // helper to derive the subcommand
     function deriveSubcommand(namespace) {
       const match = namespace.match(/[^:]+:(.+)/);
       return match ? match[1] : namespace;
     }
 
-    let defaultSubGenerator;
+    // filter the hidden subgenerators already
+    //   -> subgenerators must be found in env as they are returned by lookup!
+    let subGenerators = env
+      .lookup({ localOnly: true, packagePaths: generatorPath })
+      .filter((sub) => {
+        const subGenerator = env.get(sub.namespace);
+        return !subGenerator.hidden;
+      });
 
+    // list the available subgenerators in the console (as help)
     if (this.options.list) {
-      const subgen = env
-        .lookup({ localOnly: true, packagePaths: generatorPath })
-        .filter((sub) => {
-          return !env.get(sub.namespace)?.hidden;
+      let maxLength = 0;
+      this.log(subGenerators
+        .map(sub => {
+          maxLength = Math.max(sub.namespace.length, maxLength);
+          return sub;
         })
-        .map((sub) => {
-          let line = `${deriveSubcommand(sub.namespace)}\t\t`;
-          if (deriveSubcommand(sub.namespace).length < 8) {
-            line += "\t";
+        .reduce((output, sub) => {
```

Comment:
reduce is a very effective command to convert an array into a e.g. string (finally a single value)